bentoml/_internal/io_descriptors/base.py

```diff
@@ -44,12 +54,26 @@ class IODescriptor(ABC, t.Generic[IOType]):
     _rpc_content_type: str = "application/grpc"
     _proto_fields: tuple[ProtoField]
 
+    def __init_subclass__(cls, *, descriptor_id: str | None = None):
+        if descriptor_id is not None:
+            IO_DESCRIPTOR_REGISTRY[descriptor_id] = cls
+        cls.descriptor_id = descriptor_id
```

Comment:
Should we just set the `descriptor_id` to the `{cls.__module__}.{cls.__qualname__}`?

Reply:
No, that would make it `bentoml._internal...`, and I think that shouldn't be the ID of the descriptor.

Reply:
where are you using descriptor_id other than registering to `REGISTRY`?

Reply:
It gets exposed as part of our OpenAPI spec.